### PR TITLE
saveApplication

### PR DESCRIPTION
**Pull request by knockknockyoo**
- if a new application, store status as new
- if update application, store status as in progress

### CODE REVIEW

**Review by Wesley-Lau — APPROVED** [review]

File: services/application.js [hunk 1]
```diff
@@ -45,28 +47,38 @@ class Application {
...
+            }
         }
         return application || null;
     }
 
     async createApplication(application, userInfo) {
         let newApplicationProperties = {
             _id: v4(undefined, undefined, undefined),
-            status: IN_PROGRESS,
+            status: NEW,
```
Comment (knockknockyoo) [inline services/application.js:67]: Here is the submission history

File: services/application.js [hunk 2]
```diff
@@ -45,28 +47,40 @@ class Application {
...
             applicant: {
                 applicantID: userInfo._id,
                 applicantName: userInfo.firstName + " " + userInfo.lastName,
                 applicantEmail: userInfo.email
             },
+            history: [HistoryEventBuilder.createEvent(userInfo._id, NEW, null)],
```
Comment (knockknockyoo) [inline services/application.js:75]: a new application history is added

File: services/application.js [hunk 3]
```diff
@@ -146,24 +161,28 @@ class Application {
     async submitApplication(params, context) {
         verifySession(context)
             .verifyInitialized();
-        let application = await this.getApplicationById(params._id);
+        const application = await this.getApplicationById(params._id);
         verifyApplication(application)
             .notEmpty()
-            .state(IN_PROGRESS);
+            .state([NEW, IN_PROGRESS]);
```
Comment (knockknockyoo) [inline services/application.js:167]: Please, verify if NEW status is allowed to submit an application.

File: services/application.js [hunk 4]
```diff
@@ -188,6 +215,7 @@ class Application {
         let result = null;
         if (deletedOne && await this.dbService.deleteOne(APPLICATION, {_id: document._id})) {
             result = deletedOne[0];
+            // TODO update application status and log events
```
Comment (knockknockyoo) [inline services/application.js:218]: I commented it for next pull request.

**Pull request merged**